### PR TITLE
Make sensor server code compatible with multiple sensor types (CU-86duvwbq2)

### PR DESCRIPTION
**Pull request by Nakul727**
- Updated **brave-alert-lib to v15.0.4** for deviceType enum change
- Added migration script (migrationID 53) to update device_type_enum to handle multi-stall and single-stall sensors separately.
- Updated device_type_enum to have 'DEVICE_SENSOR_SINGLESTALL' and 'DEVICE_SENSOR_MULTISTALL' instead of 'DEVICE_SENSOR'. <br></br>
- Updated the following db functions: getActiveSensorClients(), getLocations(), createLocationFromBrowserForm(), getDataForExport(), getLocationsFromClientId(), createLocation(), getRecentSensorsVitals() and getRecentSensorsVitalsWithClientId()
- Added functionality to dashboard for submit new location and edit location to handle different device_types using dropdown select 
- Updated test cases for newLocation and updateLocation to use both sensor types
- Wrote new tests for functionality such as inserting a multi-stall sensor and editing single-stall to multi-stall
<hr></hr>

Tests are correctly passing after change to device_type_enum made by the migration script. When merged to dev, the migration script should run allowing for the new test cases to pass.

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-86duvwbq2 Make server code compatible with multiple sensor types (Single-Occupancy vs Multi-stall)](https://app.clickup.com/t/86duvwbq2)

File: server/db/db.js
Comment (Darrickg) [inline server/db/db.js:331]: this function (and many others i think?) wouldn't work with singlestall only as device type. u need to do both singlestall and multistall, especially since the functions u changed are for dasboard stuff.

File: server/db/db.js
Comment (Darrickg) [inline server/db/db.js:1265]: likewise, for test cases, u will want to test some inputs with multistall and make the others compatible with both multi and single stalls.

**Review by Darrickg — CHANGES_REQUESTED** [review]

File: server/db/db.js
Comment (Nakul727) [reply server/db/db.js:331]: Yup, you're right. I will change all the database functions to handle and retrieve multistall devices as well. Along with that, with each corresponding function, I can change the test cases to use handle input for both singlestall and multistall. This would probably involve writing new test cases. Thanks!

**Comment by Nakul727** [comment]
Note to self: Make sure to test the smoke test as the createLocation function was updated to have deviceType. Currently set up so that smoke test will create a **singlestall** device in db and then work on it. No need to run smoke test for multistall separately.

File: server/db/db.js
Comment (Nakul727) [reply server/db/db.js]: Completed in latest commits

**Review by Darrickg — APPROVED** [review]
LGTM! Feel free to merge once u get Johnny's approval

**Comment by johnny-brave** [comment]
LGTM

**Pull request merged**
